sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -304,6 +304,17 @@ def _more_tags(self):
         return {"requires_positive_y": True}
 
 
+class FailsWithConstantFeatures(BaseEstimator):
+    # Estimator that fails with constant features and don't raise
+    # an adequate error message
```

Comment:
We should also have a test that makes sure the check passes when  the error message is adequate. Maybe pass that as an __init__ parameter so you can test both cases easily?